### DISCUSSION TITLE
Adding parameters to conditional '--with' in the command line

### DISCUSSION THREAD

**Discussion by dletai** [rpm-software-management/rpm#3206] [Q&A]
In configure it's often required to do something along the lines of `./configure --with-feature=path` for example
`--with-nvml=/usr/local/cuda`

Using %bcond_with mechanism, there doesn't seem to be a way to make this happen in command line, aside from something like
`rpmbuild -rb --define '_with_nvml --with-nvml=/usr/local/cuda'`
instead of the more intuitive (but not working)
`rpmbuild -rb --with nvml=/ur/local/cuda`

Is there a way (maybe with `%bcond`) to pass such arguments from intuitive command line to `%configure` (possibly with some handling by `%if %(with xxx)` construct?

**Comment by dmnks** [comment]
Indeed, there's currently [no built-in way](https://github.com/rpm-software-management/rpm/blob/ef87d2503498f65577b5d7af07cd453d622fe02c/rpmpopt.in#L194) for the `--with foo=<arg>` syntax on the CLI.

Another workaround that comes to mind (that's perhaps a bit cleaner/clearer than defining the `_with_foo` macro directly):

`rpmbuild -rb --with nvml --define 'nvml_path /usr/local/cuda'`

The in the spec file:

```
%configure \
   %{?with_nvml:--with-nvml=%{nvml_path}}
```

Although that's still not ideal, of course.
